### ISSUE TITLE
Linux RAPL failed on master branch

### ISSUE THREAD

**Issue by benoit-cty** [mlco2/codecarbon#275]
* CodeCarbon version: master at 01/01/2022
* Python version: 3.8.10
* Operating System: Ubuntu 20.04 Linux kernel 5.4

### Description

The code crash a few seconds after the launch.

### What I Did

```
# python3 examples/api_call_debug.py 
[codecarbon DEBUG @ 19:56:01] GO!
[codecarbon INFO @ 19:56:01] [setup] RAM Tracking...
[codecarbon INFO @ 19:56:01] [setup] GPU Tracking...
[codecarbon INFO @ 19:56:01] No GPU found.
[codecarbon INFO @ 19:56:01] [setup] CPU Tracking...
[codecarbon DEBUG @ 19:56:01] Not using PowerGadget, an exception occurred while instantiating IntelPowerGadget : Platform not supported by Intel Power Gadget
[codecarbon DEBUG @ 19:56:01] We will read Intel RAPL files at /sys/class/powercap/intel-rapl/intel-rapl:0/energy_uj
[codecarbon INFO @ 19:56:01] Tracking Intel CPU via RAPL interface
Traceback (most recent call last):
  File "examples/api_call_debug.py", line 37, in <module>
    model = train_model()
  File "/root/test/codecarbon/codecarbon/emissions_tracker.py", line 778, in wrapped_fn
    tracker = EmissionsTracker(
  File "/root/test/codecarbon/codecarbon/emissions_tracker.py", line 267, in __init__
    hardware = CPU.from_utils(self._output_dir, "intel_rapl")
  File "/root/test/codecarbon/codecarbon/external/hardware.py", line 182, in from_utils
    cpu = cls(output_dir=output_dir, mode=mode, model=model, tdp=tdp)
  File "/root/test/codecarbon/codecarbon/external/hardware.py", line 105, in __init__
    self._intel_interface = IntelRAPL(rapl_dir=rapl_dir)
  File "/root/test/codecarbon/codecarbon/core/cpu.py", line 160, in __init__
    self._setup_rapl()
  File "/root/test/codecarbon/codecarbon/core/cpu.py", line 167, in _setup_rapl
    if os.path.exists(self._lin_rapl_dir):
  File "/usr/lib/python3.8/genericpath.py", line 19, in exists
    os.stat(path)
TypeError: stat: path should be string, bytes, os.PathLike or integer, not NoneType


```

```
# ll /sys/class/powercap/intel-rapl/intel-rapl:0/energy_uj
-r-------- 1 root root 4096 Jan 11 20:06 /sys/class/powercap/intel-rapl/intel-rapl:0/energy_uj
```

```
# cat /sys/class/powercap/intel-rapl/intel-rapl:0/energy_uj
232936593866

```

**Comment by vict0rsch** [comment]
Cannot replicate for now as my cluster's `powercap` has disappeared over the holidays :( I opened a ticket. @mathilde-leval would you be able to have a look?

**Comment by mathilde-leval** [comment]
Hello yes I can take a look

**Comment by vict0rsch** [comment]
Update: I still can't because of the infamous security fix changing RAPL permissions :( (#244)

**Comment by benoit-cty** [comment]
Closed by  https://github.com/mlco2/codecarbon/pull/277
